### PR TITLE
Fix #2355: shifted helper function to central utility testing module

### LINKED ISSUE #2355: Shift helper functions `hasErrorText` and `hasNoErrorText` to central utility testing file
**Is your feature request related to a problem? Please describe.**
Helper functions `hasErrorText` and `hasNoErrorText` is being used frequently in test files like `AdminAuthActivityTest` and it will be used in more files in future. 

**Describe the solution you'd like**
We should shift these to a central file rather than including them every time in the test files.

### PR DESCRIPTION
**Pull request by Akshat-gour**
## Explanation
This PR Fixes #2355: In which we Shifted helper function **hasErrorText** and **hasNoErrorText** from **org.oppia.android.app(test)** > **profile** > ( **AddProfileActivityTest.kt** , **AdminAuthActivityTest.kt** , **AdminPinActivityTest.kt** , **PinPasswordActivityTest.kt** )
To **org.oppia.android.testing** > **EditTextInputAction.kt** and added Context Parameter to EditTextInputAction Constructor and added Material Dependency to build.gradle(:testing) and to build.bazel(testing)

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: testing/BUILD.bazel [hunk 1]
```diff
@@ -27,6 +27,7 @@ kt_android_library(
         artifact("com.google.truth:truth"),
         artifact("androidx.test.ext:junit"),
         artifact("org.mockito:mockito-core"),
+        artifact('com.google.android.material:material:1.2.0-alpha02'),
         artifact("junit:junit:4.12"),
```
Comment (prayutsu) [inline testing/BUILD.bazel:43]: Add this according to its lexicographical order.

File: testing/build.gradle [hunk 2]
```diff
@@ -54,6 +54,7 @@ dependencies {
       'androidx.test:runner:1.2.0',
       'com.google.dagger:dagger:2.24',
       'org.jetbrains.kotlinx:kotlinx-coroutines-test:1.2.2',
+      'com.google.android.material:material:1.2.0-alpha02',
       'org.robolectric:robolectric:4.3',
```
Comment (prayutsu) [inline testing/build.gradle]: Add this according to its lexicographical order.

File: testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt [hunk 3]
```diff
@@ -43,4 +49,30 @@ class EditTextInputAction @Inject constructor(
       }
     }
   }
+
+  fun hasErrorText(@StringRes expectedErrorTextId: Int): Matcher<View> {
```
Comment (anandwana001) [inline testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt]: Create a new utility file for these functions. Don't add them in the `EditTextInputAction.kt`, as this file is specific to EditText helper functions only.

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt (hunk 3 above)
Comment (Akshat-gour) [reply testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt]: @anandwana001 
Can i create a class name **EditInputAction.kt** inside org.oppia.android.testing
and then reference hasErrorText and hasNoErrorText inside AddProfileActivityTest.kt , AdminAuthActivityTest.kt , AdminPinActivityTest.kt , PinPasswordActivityTest.kt

Comment (anandwana001) [reply testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt]: Can we have a more understandable name which gives a sense that we are using helper function for `TextInputLayout`?

Comment (Akshat-gour) [reply testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt]: @anandwana001 
can we use HelperTextInputLayout or TextInputLayoutHelper

Comment (anandwana001) [reply testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt]: WDYT about `TextInputAction`?

Comment (Akshat-gour) [reply testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt]: @anandwana001 
yes ,do i have to create it inside org.oppia.android.testing or somewhere else

Comment (anandwana001) [reply testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt]: Here - https://github.com/oppia/oppia-android/tree/develop/testing/src/main/java/org/oppia/android/testing

This is where we keep all the helper class files.

Comment (Akshat-gour) [reply testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt]: @anandwana001 
I created TextInputLayout.kt file there are few imports which are not in use so should I remove them.

Comment (anandwana001) [reply testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt]: yes please, else you will not be able to push code as it is kotlin file formatting error.

Comment (anandwana001) [reply testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt]: Also, please resolve the conflicts.

**Comment by anandwana001** [comment]
Could you please resolve the conflict before sending pr for review?

**Comment by anandwana001** [comment]
Please complete all the installation steps:
https://github.com/oppia/oppia-android/wiki#installation

**Comment by Akshat-gour** [comment]
@anandwana001 
i was trying to import  
**'com.google.android.material:material:1.2.0-alpha02'** 
in Build.bazel but here they are in different form so how to write it

**Comment by anandwana001** [comment]
@Akshat-gour 
Please check `oppia-android/third_party/versions.bzl` file. 

Example:
1. Added dependency of `livedata ktx` - https://github.com/oppia/oppia-android/blob/d2d55013605d318965594932b9224578cbc9a586/third_party/versions.bzl#L31
2. Using `livedata ktx` dependency in testing module in Build.bazel file - https://github.com/oppia/oppia-android/blob/d2d55013605d318965594932b9224578cbc9a586/testing/BUILD.bazel#L21

Please follow a similar pattern and make sure naming us correct. 
Read the doc for reference - https://github.com/oppia/oppia-android/blob/d2d55013605d318965594932b9224578cbc9a586/third_party/versions.bzl#L10

File: testing/BUILD.bazel (hunk 1 above)
Comment (anandwana001) [reply testing/BUILD.bazel:43]: Please remove these artifacts as they are duplicate here. 
If you want to add something new, please follow the third party format.

File: testing/src/main/java/org/oppia/android/testing/TextInputAction.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import android.view.View
+import androidx.annotation.StringRes
+import com.google.android.material.textfield.TextInputLayout
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.TypeSafeMatcher
+import javax.inject.Inject
+
+class TextInputAction @Inject constructor(
+  val context: Context
```
Comment (anandwana001) [inline testing/src/main/java/org/oppia/android/testing/TextInputAction.kt:13]: ktlint issues, please follow all the installation steps. 

https://github.com/oppia/oppia-android/wiki#installation

**Review by anandwana001 — CHANGES_REQUESTED** [review]

**Comment by Akshat-gour** [comment]
@anandwana001 
facing this issue I just created a class file **TextInputAction**
i read the documentation but could not figure out how to solve this issue
image

**Comment by anandwana001** [comment]
Please check the installation step, as it looks like you had downloaded the wrong google_checks.xml file. 
You can find the correct link in the installation step, please use it and let me know.
https://github.com/oppia/oppia-android/wiki#installation

**Comment by Akshat-gour** [comment]
@anandwana001 
that worked fine but one more thing .idea/codeStyle/project.xml
has generated we boiler code should I remove it
image

**Comment by anandwana001** [comment]
Don't include this file in the PR, also, if you are working on this PR you can open it, else you can close it and shift the discussion to wither gitter or new PR.

**Pull request closed without merging**